test/unit/alphabet/nucleotide/sam_dna16_test.cpp

```diff
@@ -25,6 +25,11 @@ INSTANTIATE_TYPED_TEST_CASE_P(sam_dna16, semi_alphabet_test, sam_dna16);
 INSTANTIATE_TYPED_TEST_CASE_P(sam_dna16, alphabet_constexpr, sam_dna16);
 INSTANTIATE_TYPED_TEST_CASE_P(sam_dna16, semi_alphabet_constexpr, sam_dna16);
 
+struct sam_dna16_accessor : public sam_dna16
+{
+    using sam_dna16::rank_to_char;
```

Comment:
so this is protected in sam_dna_16, but we have a test that tests this conversion of the sam?
Should that test not use the public interface to check the correctness of the conversion? I mean I am asking myself why it is protected and not private?